python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -127,6 +127,7 @@ def __init__(self, provider_config, cluster_name):
         self.ready_for_new_batch.set()
         self.tag_cache_lock = threading.Lock()
         self.count_lock = threading.Lock()
+        self.reuse_node_lock = threading.Lock()
```

Comment:
```suggestion
        self._reuse_node_lock = threading.Lock()
```
private field. up to you given other fields don't follow this

Reply:
I'd also drop a comment about what this specifically protects.